File: 1672/typescript/1672.ts
// tsc 1672.ts
// node 1672.js

function maximumWealth(accounts: number[][]): number {
	let maxWealth: number = 0;

	for (const customer of accounts) {
		let currentWealth: number = 0;

		currentWealth = customer.reduce((sum, balance) => sum + balance, 0);
		maxWealth = Math.max(maxWealth, currentWealth);
	}

	return maxWealth;
}

console.log(
	maximumWealth([
		[1, 2, 3],
		[3, 2, 1],
	])
);

console.log(
	maximumWealth([
		[1, 5],
		[7, 3],
		[3, 5],
	])
);

File: 1672/python/1672.py
# python 1672.py

class Solution(object):
    def maximumWealth(self, accounts):
        return max(sum(account) for account in accounts)

solution = Solution()

op1 = [[1, 2, 3], [3, 2, 1]]
op2 = [[1, 5], [7, 3], [3, 5]]

print(solution.maximumWealth(op1))
print(solution.maximumWealth(op2))
